aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,319 @@
+
+.. image:: filters/_static/filter-pipeline-2.png
+
+Selecting a Filter
+^^^^^^^^^^^^^^^^^^
+
+:ref:`Authorization filters <authorization-filters>` are used to determine whether the current user is authorized for the request being made.
+
+:ref:`Resource filters <resource-filters>` are the first filter to handle a request after authorization, and the last one to touch the request as it is leaving the filter pipeline. They're especially useful to implement caching or otherwise short-circuit the filter pipeline for performance reasons.
+
+:ref:`Action filters <action-filters>` wrap calls to individual action method calls, and can manipulate the arguments passed into an action as well as the response returned from it.
+
+:ref:`Exception filters <exception-filters>` are used to apply global policies to unhandled exceptions in the MVC app.
+
+:ref:`Result filters <result-filters>` wrap the execution of individual action results, but only when no exceptions have been thrown. They are ideal for logic that must surround view execution or formatter execution.
+
+Implementation
+^^^^^^^^^^^^^^
+
+All filters support both synchronous and asynchronous implementations through different interface definitions. Choose the sync or async variant depending on the kind of task you need to perform. They are interchangeable from the framework's perspective.
+
+Synchronous filters define both an On\ *Stage*\ Executing and On\ *Stage*\ Executed method (with noted exceptions). The On\ *Stage*\ Executing method will be called before the event pipeline stage by the Stage name, and the On\ *Stage*\ Executed method will be called after the pipeline stage named by the Stage name.
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/SampleActionFilter.cs
+  :language: c#
+  :emphasize-lines: 6,8,13
+
+Asynchronous filters define a single On\ *Stage*\ ExecutionAsync method that will surround execution of the pipeline stage named by Stage. The On\ *Stage*\ ExecutionAsync method is provided a *Stage*\ ExecutionDelegate delegate which will execute the pipeline stage named by Stage when invoked and awaited.
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/SampleAsyncActionFilter.cs
+  :language: c#
+  :emphasize-lines: 6,8-9
+
+.. note:: You should only implement *either* the synchronous or the async version of a filter interface, not both. If you need to perform async work in the filter, implement the async interface. Otherwise, implement the synchronous interface. The framework will check to see if the filter implements the async interface first, and if so, it will call it. If not, it will call the synchronous interface's method(s). If you were to implement both interfaces on one class, only the async method would be called by the framework. Also, it doesn't matter whether your action is async or not, your filters can be synchronous or async independent of the action.
+
+Filters and Attributes
```

Comment:
Let's rename this section to "Filter Scopes" and move the content about global filters from the "Configuring Filters" section to here. I would leave the stuff about global filters and DI (adding a filter by type) in the DI section and just talk about adding global filter instances here.